api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters/solutions-output/result-parser.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@nestjs/common';
+import { plainToClass } from 'class-transformer';
+import { ExecutionResult, ResultRow } from '@marxan/marxan-output';
+import { isDefined } from '@marxan/utils';
+import { validateSync } from 'class-validator';
+
+@Injectable()
+export class ResultParserService {
+  parse(csvContent: string): ExecutionResult {
+    return csvContent
+      .split('\n')
+      .slice(1)
+      .map((row) => {
+        if (row === '') {
+          return;
+        }
+        const [
+          runId,
+          score,
+          cost,
+          planningUnits,
+          connectivity,
+          connectivityTotal,
+          connectivityIn,
+          connectivityEdge,
+          connectivityOut,
+          connectivityInFraction,
+          penalty,
+          shortfall,
+          missingValues,
+          mpm,
+        ] = row.split(',');
+        const entry = plainToClass<ResultRow, ResultRow>(ResultRow, {
+          runId: +runId,
+          score: +score,
+          cost: +cost,
+          planningUnits: +planningUnits,
+          connectivity: +connectivity,
+          connectivityTotal: +connectivityTotal,
+          connectivityIn: +connectivityIn,
+          connectivityEdge: +connectivityEdge,
+          connectivityOut: +connectivityOut,
+          connectivityInFraction: +connectivityInFraction,
+          penalty: +penalty,
+          shortfall: +shortfall,
+          missingValues: +missingValues,
+          mpm: +mpm,
+          best: false,
+          mostDifferent: false,
+        });
+        if (validateSync(entry).length > 0) {
+          throw new Error(`Unexpected values in Marxan output: [${row}]`);
```

Comment:
Assuming that the marxan cli outputs reasonably validated data we may not see many of these errors, but just in case, I suggest to include the iteration index here, to make things easier to debug if any such error is raised (and propagated to the API consumer)

Reply:
@hotzevzl iteration index (runId) is already there - I can add the array's item index tho

Reply:
yep, that's what I meant, the `map()` iteration, not the Marxan one, sorry for the ambiguous wording 😅  and thanks for adding this.